docs/user-guides/best-practices.md

```diff
@@ -0,0 +1,162 @@
+# Best Practices
+
+## Inline vs Runner Scripts
+
+[Runner scripts](script-constructors.md#runner-scripts) have a stronger feature set than inline scripts so are the
+recommended path. You will likely need to set up a CICD solution as the iterative development process can be cumbersome.
+Read more in [Iterating on a Workflow](#iterating-on-a-workflow) and [CICD](#cicd).
+
+## How To Write Workflows
+
+### Laying Out Your Python Code
+
+Hera aims to keep orchestration code outside of your business logic, meaning you should write business logic in
+functions decorated with `@script`, and construct your Workflow separately, as seen throughout the examples. This might
+involve putting code in different submodules in your application, but overall you are free to write your Workflow how
+you want.
+
+An example directory structure for a project using `poetry` might be:
+
+```console
+.
+├── Dockerfile
+├── hera_scratch
+│   ├── __main__.py
+│   └── workflow.py
+├── Makefile
+├── poetry.lock
+├── pyproject.toml
+├── README.md
+└── requirements.txt
```

Comment:
I thought Poetry put requirements in `pyproject.toml`?

Reply:
Yes, the `requirements.txt` is exported for docker. It comes from a `make` target here:

https://github.com/elliotgunton/hera-example-project/blob/e01a2f3a13dab907f659a4f20dcda7fa6cce6c7f/Makefile#L7